pkg/kubelet/kubelet_pods.go

```diff
@@ -85,6 +86,8 @@ const (
 	kubeletUser = "kubelet"
 )
 
+const sysctlAllDisableIPv6 = "/proc/sys/net/ipv6/conf/all/disable_ipv6"
```

Comment:
I think this moved the wrong direction, but I'm not really as deep in SIG node as I once was.  I feel like this SHOULD be passing a `sysctl.Interface` to be used, rather than hard-coding a path and doing manual parsing.  It's rather inconsequential since we don't have tests for `ensureHostsFile`.  We shouldm and if we did we would not be able to just read from /proc.

@mrunalp @SergeyKanzhelev how much do you care about this?  It's minor in the grand scheme of things.

I suggest one of two paths:

1) Go back to `utilsysctl.New().GetSysctl()` and leave a "TODO" comment that this should be injected by the caller.

2) Create a kubelet-level `sysctl.Interface` early, store it, and pass it as an argument to `ensureHostsFile` (or make that a method which reads the Kubelet.sysctl.

Reply:
This could be a fun starter project for someone - find all the places in kubelet that use sysctl and make them all use an injected `sysctl.Interface`, which is created early in Kubelet's lifecycle.

Reply:
> 2. Create a kubelet-level `sysctl.Interface` early, store it, and pass it as an argument to `ensureHostsFile` (or make that a method which reads the Kubelet.sysctl.

I think it would be better to add `sysctl.Interface` in the [kubelet struct](https://github.com/kubernetes/kubernetes/blob/06b813fd29bcf5f69184259b3b07af612b70077b/pkg/kubelet/kubelet.go#L953-L954) and pass it on, although we need to span `makeMounts` -> `makeHostsMount` -> `ensureHostsFile` -> `managedHostsFileContent`......



> find all the places in kubelet that use sysctl and make them all use an injected `sysctl.Interface`

We currently only use it in [setupKernelTunables](https://github.com/kubernetes/kubernetes/blob/06b813fd29bcf5f69184259b3b07af612b70077b/pkg/kubelet/cm/container_manager_linux.go#L412).

Reply:
> I think it would be better to add ...

I agree, though after writing it, I am not sure that should be in THIS PR.  I'd like someone more involved in kubelet to decide if that's important enough to do right now

Reply:
Okay, let's keep it until sig-node's approver makes the decision.

Reply:
I think the previous form (utilsysctl.New) is better than this, and the change to injected sysctl.Interface will be simpler

Reply:
reverted.

Reply:
ping @mrunalp @SergeyKanzhelev